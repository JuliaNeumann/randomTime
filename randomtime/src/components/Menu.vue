<template>
  <div class="uk-margin">
    <nav class="uk-navbar-container" uk-navbar>
      <div class="uk-navbar-right">

        <ul class="uk-navbar-nav">
          <li><router-link to="weekplan">Create Week Plan</router-link></li>
          <li><router-link to="dayplan">Get Day Plan</router-link></li>
        </ul>

      </div>
    </nav>
  </div>
</template>

<script>
export default {
  name: 'Menu',
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>

</style>
