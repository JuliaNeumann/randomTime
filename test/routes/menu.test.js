import { shallow } from 'vue-test-utils';
import Menu from '../../app/routes/main/menu.vue';

describe('Menu', () => {
  const wrapper = shallow(Menu, {
    propsData: {
      active: 'weekplan'
    }
  });

  it('renders the correct markup', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('marks the current page menu item as active', () => {
    const menuItems = wrapper.findAll('li');
    expect(menuItems.at(0).classes()).toContain('uk-active');
    expect(menuItems.at(1).classes()).not.toContain('uk-active');
  });
});