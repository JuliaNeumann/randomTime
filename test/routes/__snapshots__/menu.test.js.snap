// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders the correct markup 1`] = `
<div
  class="uk-margin"
>
  <nav
    class="uk-navbar-container"
    uk-navbar=""
  >
    <div
      class="uk-navbar-right"
    >
      <ul
        class="uk-navbar-nav"
      >
        <li
          class="uk-active"
        >
          <a
            href="/weekplan"
          >
            Create Week Plan
          </a>
        </li>
         
        <li
          class=""
        >
          <a
            href="/dayplan"
          >
            Get Day Plan
          </a>
        </li>
         
        <li
          class=""
        >
          <a
            href="/config"
          >
            Config
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
